pkg/migrations/sql/idempotent/base.sql

```diff
@@ -72,26 +72,34 @@ CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.lock_metric_for_maintenance(metric_id
     RETURNS BOOLEAN
 AS $func$
 DECLARE
+    res BOOLEAN;
 BEGIN
     IF NOT wait THEN
-        PERFORM m.*
-        FROM SCHEMA_CATALOG.metric m
-        WHERE m.id = metric_id
-        FOR NO KEY UPDATE SKIP LOCKED;
+        SELECT pg_try_advisory_lock(12378, metric_id) INTO STRICT res;
 
-        RETURN FOUND;
+        RETURN res;
     ELSE
-        PERFORM m.*
-        FROM SCHEMA_CATALOG.metric m
-        WHERE m.id = metric_id
-        FOR NO KEY UPDATE;
+        PERFORM pg_advisory_lock(12378, metric_id);
```

Comment:
Having a note basically everywhere we use this 12378 thing is probably a good idea, it strikes me as odd every time I see it.

Reply:
I think this should be done like how we did below

https://github.com/timescale/promscale/blob/9c6e713a96cdb93154cf319702c52619913e8bd6/pkg/pgmodel/migrate.go#L372